docs/pages/blog/masonry-post.md

```diff
@@ -0,0 +1,141 @@
+- span multiple columns,
+- insert each item in the shortest column.
+
+![Screenshot 2021-09-15 at 09 29 54](https://user-images.githubusercontent.com/32841130/133398931-a9ee1aa8-11ca-47e7-a356-d051dab30435.png)
+
+![Screenshot 2021-09-15 at 09 29 59](https://user-images.githubusercontent.com/32841130/133398945-0e3ea284-8d67-47a3-afad-36a95a6f6925.png)
+
+![Screenshot 2021-09-15 at 09 30 06](https://user-images.githubusercontent.com/32841130/133398964-059e0faf-d624-472d-8fbb-9b1ede41cce6.png)
+
+With the help of these survey results, we came up with the following requirements:
+
+**Must-have**:
+
+- Items can have arbitrary heights.
+- Items should flow horizontally.
+- Each item is pushed to the shortest column.
+
+**Should-have**:
+
+- The number of columns, `columns`, and the spacing between items, `spacing`, can be passed into the component.
+- Responsive number of columns (breakpoints)
+- Responsive spacing (breakpoints)
+
+**Nice-to-have**:
+
+- Server side rendering
+- Span multiple columns in a masonry
+- Performance
+- Logical tab order
+- Support for lazy loading/virtualisation
+
+In short, we wanted to build a masonry component that is light-weight enough to be performant and easy-to-use, but solid enough to do its job.
+
+## Two possible solutions: CSS Grid or CSS Flexbox?
+
+Since our goal was to develop a light-weight masonry component, an absolute positioning implementation, which most existing masonry libraries make use of, was not an option. It requires too much JavaScript, and therefore it is likely to be heavy-weight.
+
+This left either CSS Grid or CSS Flexbox, but each came with its own advantages and disadvantages.
+
+**1. CSS Grid + Configuring `grid-row-start` or `grid-row-end`of items**
+
+The idea is to create a grid container, `<Masonry />`, that contains grid items, `<MasonryItem />`s, each of which contains a child. Then, we compute the height of each child and set the height of its wrapper, `<MasonryItem />`, to this value. This allows every `<MasonryItem />` to be able to have a different height and to be large enough to display its child.
+
+One important thing to note is that simply setting a fixed height to `<MasonryItem />` will not do anything. Because it is a CSS grid item, it is constrained to the row height configured in its grid container, `<Masonry />`. So, we should use a property called `grid-row-start` or `grid-row-end`. Both properties span the height of a grid item to a specific number of rows. For example, applying `grid-row-end: span 10` to a `<MasonryItem />` spans its height to amount to 10 rows.
+
+This means that a `<MasonryItem />`'s height is always a multiple of the row height configured in `<Masonry />`. If the row height is `10px`, the height of `<MasonryItem />` is a multiple of `10px`: `100px`, `110px`, ... and so on. This limitation gives rise to a serious problem. However, there is always a solution, and both the problem and the solution are introduced in the next section.
```

Comment:
```suggestion
This means that a `<MasonryItem />`'s height is always a multiple of the row height configured in `<Masonry />`. If the row height is `10px`, the height of `<MasonryItem />` is a multiple of `10px`: `100px`, `110px`, ... and so on. This limitation rises a serious problem. However, there is always a solution, and both the problem and the solution are introduced in the next section.
```

Would be nice to explain what is the problem

Reply:
As said, both the problem and the solution are introduced in the next section. I think it is a good idea to link to the section.

Reply:
"gives rise to" is correct. "causes", while less precise, might be easier for an international audience to follow.

You might mean "raises", but that means to draw attention to a (perceived) problem.